### ISSUE TITLE
Fatal error ParseTree.php#111

### ISSUE THREAD

**Issue by pavarnos** [vimeo/psalm#165]
Hi

I see a fatal error message (below) when running psalm on my code, but there is no indication of what is causing the problem.
```
Fatal error: Uncaught Psalm\Exception\TypeParseTreeException: Cannot parse comma in non-generic/array type in /Users/stevep/Documents/overseas/isv/Personnel Database/l2s/vendor/vimeo/psalm/src/Psalm/Type/ParseTree.php:111
Stack trace:
#0 /Users/stevep/Documents/overseas/isv/Personnel Database/l2s/vendor/vimeo/psalm/src/Psalm/Type.php(52): Psalm\Type\ParseTree::createFromTokens(Array)
#1 /Users/stevep/Documents/overseas/isv/Personnel Database/l2s/vendor/vimeo/psalm/src/Psalm/Checker/FunctionLikeChecker.php(776): Psalm\Type::parseString('array(firstpers...')
#2 /Users/stevep/Documents/overseas/isv/Personnel Database/l2s/vendor/vimeo/psalm/src/Psalm/Checker/ClassLikeChecker.php(943): Psalm\Checker\FunctionLikeChecker::register(Object(PhpParser\Node\Stmt\ClassMethod), Object(Psalm\Checker\ClassChecker))
#3 /Users/stevep/Documents/overseas/isv/Personnel Database/l2s/vendor/vimeo/psalm/src/Psalm/Checker/ClassLikeChecker.php(357): Psalm\Checker\ClassLikeChecker->visitClassMethod(Object(PhpParser\Node\Stmt\ClassMethod), Object(Ps in /Users/stevep/Documents/overseas/isv/Personnel Database/l2s/vendor/vimeo/psalm/src/Psalm/Type/ParseTree.php on line 111
```

Running with --debug shows me the file name. According to phpStorm the file syntax is valid php7.1. I tried debugging but cannot seem to get it to stop on a breakpoint at or before the line. What else could i use to diagnose the problem?

**Comment by muglug** [comment]
Seems like it's trying to parse a docblock that contains the string `array(firstpers` – is that possible?

**Comment by pavarnos** [comment]
i already greped the entire code base for 'firstpers' but did not find it

**Comment by muglug** [comment]
Get latest master, I've just added specific issue about for that return type

**Comment by pavarnos** [comment]
Ah! The problem was ` @return array( first person id, second person id )`. Did not guess that psalm was eating the whitespace.
Thanks for super quick help! My bad for the invalid docblock...

**Comment by muglug** [comment]
No problems! Glad this was easy to resolve!

**Comment by pavarnos** [comment]
Another small issue?
`@param` on a line by itself results in a php notice on stderr
```
PHP Notice:  Uninitialized string offset: 0 in /Users/stevep/Documents/overseas/isv/Personnel Database/l2s/vendor/vimeo/psalm/src/Psalm/Checker/CommentChecker.php on line 172
```

no hurry to fix this one

**Comment by muglug** [comment]
Fixed in fa46789
